pallets/utility/src/benchmarking.rs

```diff
@@ -0,0 +1,217 @@
+}
+
+#[cfg(feature = "std")]
+fn make_relay_tx_users<T: Trait>() -> (User<T>, User<T>) {
+    let alice = User::<T>::new("Caller", 1);
+    let bob = User::<T>::new("Target", 1);
+
+    (alice, bob)
+}
+
+#[cfg(not(feature = "std"))]
+fn make_relay_tx_users<T: Trait>() -> (User<T>, User<T>) {
+    // Keys generated
+    let alice_pk = hex!("6a4f597d1a0004ee6fd08622baf93fc350c048aeb8a6bf253208b1a536539333");
+    let bob_pk = hex!("8a2f30f00294ca72f2e9572263c8cf96695a1d9fffff3f8b0d49171a917d9f31");
+
+    // Create account from generated keys.
+    let alice = User::<T>::new_from_public(alice_pk);
+    let bob = User::<T>::new_from_public(bob_pk);
+
+    (alice, bob)
+}
+
+fn remark_call_builder<T: Trait>(
+    signer: &User<T>,
+    _: T::AccountId,
+) -> (UniqueCall<<T as Trait>::Call>, Vec<u8>) {
+    let call = make_calls::<T>(1).pop().unwrap();
+    let nonce: AuthorizationNonce = Module::<T>::nonce(signer.account.clone());
+    let call = UniqueCall::new(nonce, call);
+
+    #[cfg(feature = "std")]
+    let encoded = {
+        // Signer signs the relay call.
+        // NB: Decode as T::OffChainSignature because there is not type constraints in
+        // `T::OffChainSignature` to limit it.
+        let raw_signature: [u8; 64] = signer.sign(&call.encode()).to_bytes();
+        let encoded = MultiSignature::from(Signature::from_raw(raw_signature)).encode();
+
+        // Native execution can generate a hard-coded signature using the following code:
+        // ```ignore
+        // let hex_encoded = hex::encode(&encoded);
+        // frame_support::debug::info!("Signer nonce:{} encoded:{:?}", nonce, &hex_encoded);
+        //  ```
+
+        encoded
+    };
+    #[cfg(not(feature = "std"))]
+    let encoded = hex!("01d6dda327f7ab364e0a6c3aa8db761c796073efe820574b2564672c99bfbdfb129dd9505a770b03161182f29d3c6e44a63a589eb3357e94644e9a7a285add8c8e").to_vec();
+
+    (call, encoded)
+}
+
+fn transfer_call_builder<T: Trait>(
+    signer: &User<T>,
+    target: T::AccountId,
+) -> (UniqueCall<<T as Trait>::Call>, Vec<u8>) {
+    let call = make_transfer_calls::<T>(1, target, 1).pop().unwrap();
+    let nonce: AuthorizationNonce = Module::<T>::nonce(signer.account.clone());
+    let call = UniqueCall::new(nonce, call);
+
+    #[cfg(feature = "std")]
+    let encoded = {
+        // Signer signs the relay call.
+        // NB: Decode as T::OffChainSignature because there is not type constraints in
+        // `T::OffChainSignature` to limit it.
+        let raw_signature: [u8; 64] = signer.sign(&call.encode()).to_bytes();
+        MultiSignature::from(Signature::from_raw(raw_signature)).encode()
+    };
+    #[cfg(not(feature = "std"))]
+    let encoded = hex!("01aa3fb75dddaa9d1c058097aa10814a46a411192124a1970e21fc9547a075045090a318672bd44baa8f2069c4484ffc1b0e133011a80a1aaf2a484970bcffd987").to_vec();
+
+    (call, encoded)
+}
+
+benchmarks! {
+    _ {}
+
+    batch {
+        let c in 1..MAX_CALLS;
```

Comment:
The case `c = 0` could also be profiled here and in below benchmarks. It's not the worst case and is arguably the best one but having it might help the benchmarking backend approximate the base weight better and not attribute the base weight to `c`.